src/openforms/forms/models/utils.py

```diff
@@ -22,11 +25,33 @@ def contribute_to_class(self, cls: ModelBase, name: str) -> None:
 
         # generate the getter function & set it on the class
         def getter(instance: models.Model) -> str:
-            value = getattr(instance, self.model_field)
+            model_field = self.model_field
+            config_field = self.config_field
+            if self.language_code:
+                model_field = f"{self.model_field}_{self.language_code}"
+                config_field = f"{self.config_field}_{self.language_code}"
+
+            value = getattr(instance, model_field)
             if value:
                 return value
 
             config = GlobalConfiguration.get_solo()
-            return getattr(config, self.config_field)
+            return getattr(config, config_field)
 
         setattr(cls, name, getter)
+
+
+def set_dynamic_literal_getters(
+    literal_names: List, cls: ModelBase, config_field_prefix: str
+) -> None:
+    """
+    Dynamically create literal getters for each available language
+    """
+    for literal in literal_names:
+        for language_code, _label in settings.LANGUAGES:
```

Comment:
is there no way to bring this `for`-loop inside the `contribute_to_class` of `literal_getter`? Then we wouldn't have to override the metaclass.

Reply:
I guess it could be possible, but I'm not sure how to access the `language_code` to the `literal_getter.contribute_to_class.getter` function, because I cannot simply pass it as an argument to `literal_getter.contribute_to_class.getter` and it would no longer be a class variable on `literal_getter` (and it doesn't have access to any variables declared in `contribute_to_class`)

Reply:
I refactored it in dc266fea76d7b233e242c07f66dfa6665d286b68 :)